mfr/server/static/js/mfr.js

```diff
@@ -75,26 +75,61 @@
-
-            self.pymParent.el.appendChild(self.spinner);
-            $(self.pymParent.iframe).on('load', function () {
-                self.pymParent.el.removeChild(self.spinner);
-            });
-
-            self.pymParent.onMessage('embed', function(message) {
-                _addClass(self.pymParent.el, 'embed-responsive');
-                _addClass(self.pymParent.el, message);
-                _addClass(self.pymParent.iframe, 'embed-responsive-item');
-            });
-
-            self.pymParent.onMessage('location', function(message) {
-                window.location = message;
-            });
+
+            const wb_url = window.contextVars.waterbutlerURL;
+            const node_id = window.contextVars.node.id;
+            const provider = window.contextVars.file.provider;
+            const file_id = window.contextVars.file.id;
+            const file_name = window.contextVars.file.name;
+
+            const wb_file_url = `${wb_url}/v1/resources/${node_id}/providers/${provider}/${file_id}`;
+
+            const handlers = {
+
+                default: function() {
+
+                    self.pymParent = new pym.Parent(self.id, self.url, self.config);
+                    self.pymParent.iframe.setAttribute('allowfullscreen', '');
+                    self.pymParent.iframe.setAttribute('webkitallowfullscreen', '');
+                    self.pymParent.iframe.setAttribute('scrolling', 'yes');
+                    self.pymParent.iframe.setAttribute('sandbox', 'allow-scripts allow-popups allow-same-origin');
+
+                    self.pymParent.el.appendChild(self.spinner);
+                    $(self.pymParent.iframe).on('load', function () {
+                        self.pymParent.el.removeChild(self.spinner);
+                    });
+
+                    self.pymParent.onMessage('embed', function(message) {
+                        _addClass(self.pymParent.el, 'embed-responsive');
+                        _addClass(self.pymParent.el, message);
+                        _addClass(self.pymParent.iframe, 'embed-responsive-item');
+                    });
+
+                    self.pymParent.onMessage('location', function(message) {
+                        window.location = message;
+                    });
+                },
+
+                html: function() {
+                    var xhr = new XMLHttpRequest()
+                    xhr.open('GET', uri, true);
+                    xhr.withCredentials = true;
+                    xhr.onreadystatechange = function() {
+                        if (xhr.readyState === 4 && xhr.status === 200) {
+                            const frame = document.createElement("iframe");
+                            frame.id = "mfrframe";
+                            document.getElementById("mfrIframe").appendChild(frame);
+                            document.getElementById("mfrframe").src = "data:text/html;charset=utf-8," + escape(xhr.responseText);
```

Comment:
If there are issues using a data url, we also have the srcdoc attribute available https://www.w3schools.com/tags/att_iframe_srcdoc.asp